tools/fetch_encoded.c

```diff
@@ -0,0 +1,106 @@
+/* Copyright (c) the JPEG XL Project Authors. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ *
+ *
+ * Auxiliary functions for fetching and processing JPEG XL decoder
+ * output that might be useful in a no-C++ context, such as when
+ * implementing a JPEG XL library via some other language's C foreign
+ * function interface.
+ */
+#include "fetch_encoded.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "jxl/encode.h"
+
+#define DEBUG_BUFFER_REALLOCATION 0
+
+JXL_BOOL fetch_jxl_encoded_image(JxlEncoder *jxl_encoder,
```

Comment:
The naming convention is different from the rest of the codebase.

Why can't we use C++ here?

If we do use C, why not `realloc`?

Reply:
1. Why not use C++: this is generic stuff that one has to also do when e.g. writing a FFI for JXL (say, for Python, Perl, Racket, nodejs, etc.). It's more useful to have this in C and allow carrying forward this code rather than forcing whoever wanted to use this for a FFI to also introduce a dependency on a C++ compiler.

2. realloc() is definitely more appropriate here - also since malloc() may have done pre-allocation. I think I just overlooked that because I was too occupied with keeping all this "how to wire up things" state in my head when this was all new to me.

Reply:
I don't think tools/ is the correct place for basically a wrapper over the API that makes it so that output is no longer streaming (if I understand correctly). I'd argue that someone writing a FFI layer should still expose the streaming output mode of the C API, so I'm not sure there's a point to make this not c/djxl-private.

Reply:
Just as a proposal, I just submitted a version that does not use `fetch_jxl_encoded_image`, `write_jxl_file` nor the `ManagedJxlEncoder` (also to address the first unresolved comment). This now basically follows how it is done in `encode_oneshot.cc`. 
Not sure how this plays with a potential FFI, but perhaps this can be done at another place.

Reply:
decided to remove `fetch_encoded.*` and instead go with a custom `WriteFile`